xbmc/cores/RetroPlayer/rendering/RenderContext.cpp

```diff
@@ -139,7 +143,8 @@ int CRenderContext::GUIShaderGetPos()
   CRenderSystemGL* renderingGL = dynamic_cast<CRenderSystemGL*>(m_rendering);
   if (renderingGL != nullptr)
     return static_cast<int>(renderingGL->ShaderGetPos());
-#elif HAS_GLES >= 2
+#endif
+#if HAS_GLES >= 2
```

Comment:
About here have you a link how it becomes handled on end?
Looks a bit critical if both `HAS_GL` and `HAS_GLES` becomes for now defined (by not expected thing) and the `dynamic_cast` used on both with `m_rendering`.

Reply:
Yes unfortunately this was the easiest option at the time to get this POC code running. Basically this commit shows where our abstraction is lacking in terms of render system inheritance. I would like to clean this up on the future and move the `dynamic_cast` to proper inheritance with derived classes.

You can click this link to see the commit without the white space changes as it's much easier to understand what the intention is.

https://github.com/xbmc/xbmc/commit/71a4aa2e90f6a517ed0f44364c6ab3d95c70d072?w=1